common/lib/dependabot/pull_request_creator/branch_namer.rb

```diff
@@ -37,7 +38,14 @@ def new_branch_name
           end
 
         # Some users need branch names without slashes
-        sanitize_ref(File.join(prefixes, @name).gsub("/", separator))
+        sanitized_name = sanitize_ref(File.join(prefixes, @name).gsub("/", separator))
+
+        # Shorten the ref in case users refs have length limits
+        if @max_length
+          sanitized_name[0, @max_length]
```

Comment:
I'm worried that we'd end up with duplicate branch names in this scenario, as we'd end up truncating the bit that makes them unique (the version), and I think that could cause issues

Reply:
That's a valid concern.

Would you be okay with taking a hash of the full-length branch name and taking the first `max_length` characters from that if `max_length` is specified? (At some point we'd have to choose the lesser of 2 evils, we cannot deal with going over the limit, however then you do lose pretty branch names)

Reply:
Yeah a hash could also work, it's a tad safer I guess (if for some reason the combination of dependency names changes, it's still unique)

Reply:
Very clean solution @TomNaessens 👍